### ISSUE TITLE
Questions about hyp.scratch and hyp.finetune

### ISSUE THREAD

**Issue by isomatical** [ultralytics/yolov5#3689]
## ❔Question
Hi, could anyone tell me which yaml should I choose when training a custom dataset?
My understanding is that:
1. if I load a pretrained weights like yolov5s.pt into the model, I should use hyp.finetune. If I use hyp.scratch, since the initial learning rate is relatively large (0.01), the pretrained weights would be kind of messed up in early epochs, thus making it meaningless to use pretrained weights. 
2. hyp.scratch is used to train large datasets like coco from scratch. For small custom datasets, training from scratch won't get good results.
Am I correct?

**Comment by glenn-jocher** [comment]
@isomatical 

- hyp.scratch.yaml has been evolved to train COCO from scratch
- hyp.finetune.yaml has been evolved to train VOC from pretrained COCO weights. 

You can find more information directly in the hyp files:
https://github.com/ultralytics/yolov5/blob/bf209f6fe92abc4c0ea66a549f6527c7fe8ec20a/data/hyp.finetune.yaml#L1-L10

Using either file with pretrained weights will produce faster results than training from --weights '', though as you mentioned a high LR in scratch.yaml will cause an early dip in accuracy. 

For small datasets it is always recommended to start training from pretrained COCO weights. You can see our full list of recommendations in **Tips for Best Training Results** below:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by isomatical** [comment]
Thanks! Another question please:
Is there a rule of thumb for setting --epochs when training a small datasets (e.g. 2000 training datas) with the pretrained COCO weights and hyp.scratch yaml. 
To my knowledge, when using pretrained COCO weights to finetune custom datasets, great results could be generated in no more than 10 epochs because the optimization is mostly for the head part (the weights of backbone are relatively fixed and stable because it is already capable of extracting features after training on COCO.)
Correct me if I am wrong, thanks in advance.

**Comment by glenn-jocher** [comment]
@isomatical start at 300 and modify as you see fit based on those training results.

**Comment by isomatical** [comment]
Hi glenn, you mentioned this in the hyperparameter evolution tutorial.
![image](https://user-images.githubusercontent.com/37702903/122726485-9f119380-d2a8-11eb-8a8b-0f5e34297b83.png)
I used the second command line but it looked like the hyp.scratch yaml was automatically used rather than the hyp.finetune.yaml as I lookd through the hyp.yaml generated in my project dir. 
And in train.py, the default value for --hyp is  'data/hyp.scratch.yaml' . 
So could you clarify it once again?

**Comment by glenn-jocher** [comment]
@isomatical default value is here:
https://github.com/ultralytics/yolov5/blob/75c0ff43af18d9d90b32ccfadd6029573b2a502a/train.py#L477

You can point to any hyp yaml with
```bash
python train.py --hyp path/to/hyp.yaml
```
